Migrations: Fix to upgrade from 0.4.0 migration

Hwesta opened the pull request:
Fix 0004_v0_4_0 migration to match what's in the stable/0.4.0 branch. Update the fedora and duracloud migrations to upgrade properly.  All generated using `manage.py schemamigration locations --auto <name>`

Hwesta committed:
Migrations: Fix to upgrade from 0.4.0 migration

Fix 0004_v0_4_0 migration to match what's in the stable/0.4.0 branch.
Update the fedora and duracloud migrations to upgrade properly.

jhsimpson committed:
Preparing for 0.5.0 release

Updating version number in setup.py. Vendor Sword2 library.

Sword2 is listed in requirements as a github uri, which doesn't work from a
launchpad build server.  Add the sword2 tarball to lib and update the
requirements file to point to it.

jhsimpson commented:
I have tested this twice and the migration from 0.4.0 to 0.5.0 worked both times

:shipit:

mistydemeo commented on requirements/production.txt:
```diff
@@ -2,4 +2,6 @@
 -r base.txt
 
+# Install from local in packaging
+Sword2
```
Let's check the tarball we're using into version control.

jhsimpson committed:
Updating debian changelog and adding requirement

httplib2 is a dependency of sword2, and needs to be
added explicitly to the requirements file of the storage service
because sword2 source is included in the lib dir now, not downloaded.

jhsimpson commented:
:shipit:
